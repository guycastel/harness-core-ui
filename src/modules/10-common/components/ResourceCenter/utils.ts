/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import moment from 'moment'
import type { AccountDTO, UserInfo } from 'services/cd-ng'

export const ON_PREM_RELEASE_NOTE_LINK = 'https://ngdocs.harness.io/article/556wy85kbo-harness-on-prem-release-notes'

export const HARNESS_SEARCH_LINK = 'https://harness.io/search/'
export const HARNESS_UNIVERISITY_LINK = 'https://developer.harness.io/certifications/'
export const HARNESS_COMMUNITY_LINK = 'https://community.harness.io/'
export const HARNESS_COMMUNITY_SLACK_LINK =
  'https://join.slack.com/t/harnesscommunity/shared_invite/zt-y4hdqh7p-RVuEQyIl5Hcx4Ck8VCvzBw'
export const HARNESS_DOCS_LINK = 'https://docs.harness.io/'
export const HARNESS_API_DOCS_LINK = 'https://apidocs.harness.io/'
export const HARNESS_TUTORIALS = 'https://developer.harness.io/tutorials/get-started'
export const SITE_STATUS_LINK = 'https://status.harness.io/'
export const HARNESS_JFROG =
  'https://developer.harness.io/docs/continuous-integration/use-ci/build-and-upload-artifacts/upload-artifacts-to-jfrog'
export const HARNESS_SUPPORT_PAGE = 'https://www.harness.io/support-search'

export const timestamp = moment.now()
export const HARNESS_SUPPORT_LINK =
  '/sso.html?action=login&brand_id=114095000394&locale_id=1&return_to=https%3A%2F%2Fsupport.harness.io%2Fhc%2Fen-us%2Frequests&src=zendesk&timestamp=' +
  timestamp
export const CANNY_SUPPORT_LINK = 'https://ideas.harness.io/'
export const WHATS_NEW_LINK = `https://docs.harness.io/article/ueeiji09do-what-s-new`
export const EARLY_ACCESS_LINK = `https://docs.harness.io/article/w4krvu96i3-early-access`

export const openZendeskSupport = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(HARNESS_SUPPORT_LINK)
}

const getInitials = (name: string) => {
  const parts = name.split('.')
  let initials = ''
  for (let i = 0; i < parts.length; i++) {
    if (parts[i].length > 0 && parts[i] !== '') {
      initials += parts[i][0]
    }
  }
  return initials
}
const initCanny = (currentUserInfo: UserInfo, account?: AccountDTO) => {
  Canny('identify', {
    appID: window.cannyAppId,
    user: {
      companies: [
        {
          id: account?.identifier,
          name: account?.companyName
        }
      ],
      email: currentUserInfo.email,
      id: currentUserInfo.uuid,
      name: account?.cannyUsernameAbbreviationEnabled
        ? currentUserInfo.name?.includes('@')
          ? getInitials(currentUserInfo.name.split('@')[0] || '') || ''
          : getInitials(currentUserInfo.name || '') || ''
        : currentUserInfo.name || ''
    }
  })
}

export const cannySupportShareYourIdeas = (
  e: React.MouseEvent<Element, MouseEvent>,
  currentUserInfo: UserInfo,
  account?: AccountDTO
): void => {
  e.stopPropagation()
  e.preventDefault()
  initCanny(currentUserInfo, account)
  window.open(CANNY_SUPPORT_LINK)
}

export const openWhatsNew = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(WHATS_NEW_LINK)
}

export const openEarlyAccess = (e: React.MouseEvent<Element, MouseEvent>): void => {
  e.stopPropagation()
  e.preventDefault()
  window.open(EARLY_ACCESS_LINK)
}

export const getReleaseNotesLink = () => {
  return window.releaseNotesLink || ON_PREM_RELEASE_NOTE_LINK
}
