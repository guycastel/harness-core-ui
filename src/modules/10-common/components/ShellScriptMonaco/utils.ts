/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

const EXPRESSION_START_REGEX = /<\+([A-Za-z0-9_.'"()]*?)$/

export function getStartColumnForMonacoRange(prevText?: string): number | undefined {
  if (typeof prevText !== 'string') return undefined
  const startColumnWithTriggerChar = prevText.match(EXPRESSION_START_REGEX)?.index
  if (startColumnWithTriggerChar === undefined) return undefined
  const startColumn = startColumnWithTriggerChar + 2 + 1 // 2 for trigger chars + 1 for monaco as it starts index from 1
  return startColumn
}
