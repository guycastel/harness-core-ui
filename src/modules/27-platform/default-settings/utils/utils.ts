/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { StoreMetadata, StoreType } from '@common/constants/GitSyncTypes'
import { SettingType } from '@common/constants/Utils'
import { ResponseListSettingResponseDTO } from 'services/cd-ng'

export const AidaAgreementType = 'AIDA'
export const defaultSettingsTagInputSeparator = ','

export const getSettingValue = (
  settingsData: ResponseListSettingResponseDTO | null,
  identifier: string
): string | undefined => {
  let settingValue: string | undefined

  if (settingsData?.data?.length) {
    settingValue = settingsData?.data?.find(element => element.setting?.identifier === identifier)?.setting?.value
  }
  return settingValue
}

export const getDefaultStoreType = (
  settingsData: ResponseListSettingResponseDTO | null
): StoreMetadata['storeType'] => {
  const settingValue = getSettingValue(settingsData, SettingType.DEFAULT_STORE_TYPE_FOR_ENTITIES)
  return settingValue === StoreType.REMOTE ? StoreType.REMOTE : StoreType.INLINE
}

/**
 * This method will include both ENFORCE_GIT_EXPERIENCE and DEFAULT_STORE_TYPE_FOR_ENTITIES settings
 * INLINE is used as fallback if no gitXSetting is defined
 * @param gitXSetting
 * @returns
 */
export function getDefaultStoreTypeFromSettings(
  gitXSetting?: ResponseListSettingResponseDTO | null
): StoreMetadata['storeType'] {
  return gitXSetting
    ? getSettingValue(gitXSetting, SettingType.ENFORCE_GIT_EXPERIENCE) === 'true'
      ? StoreType.REMOTE
      : getDefaultStoreType(gitXSetting)
    : StoreType.INLINE
}
