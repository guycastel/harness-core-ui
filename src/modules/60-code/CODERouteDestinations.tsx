/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Redirect, Switch } from 'react-router-dom'
import commonRoutes from '@common/RouteDefinitionsV2'
import { routesV2 as routes } from '@modules/60-code/RouteDefinitions'
import { useAppStore } from 'framework/AppStore/AppStoreContext'
import { Scope } from 'framework/types/types'
import { RouteWithContext } from '@common/router/RouteWithContext/RouteWithContext'
import { accountPathProps, projectPathProps, NAV_MODE, orgPathProps } from '@common/utils/routeUtils'
import { ModuleName } from 'framework/types/ModuleName'
import { PAGE_NAME } from '@common/pages/pageContext/PageName'
import { useGetSelectedScope } from '@common/navigation/SideNavV2/SideNavV2.utils'
import {
  Branches,
  Commit,
  Commits,
  Compare,
  FileEdit,
  PullRequest,
  PullRequests,
  Repositories,
  Repository,
  Search,
  Settings,
  Tags,
  WebhookDetails,
  WebhookNew,
  Webhooks
} from './CodeApp'
import { codePathProps } from './RouteDestinations'

export default function CODERouteDestinations(mode = NAV_MODE.MODULE): React.ReactElement {
  const routePaths = buildCODERoutePaths(mode)

  return (
    <Switch>
      <RouteWithContext
        exact
        path={[
          commonRoutes.toMode({ ...projectPathProps, module: CODE, mode }),
          commonRoutes.toMode({ ...orgPathProps, module: CODE, mode }),
          commonRoutes.toMode({ ...accountPathProps, module: CODE, mode })
        ]}
      >
        <CODERedirect />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODECompare} pageName={PAGE_NAME.CODEPullRequestsCompare}>
        <Compare />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODEPullRequest} pageName={PAGE_NAME.CODEPullRequests} exact>
        <PullRequest />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODEPullRequests} pageName={PAGE_NAME.CODEPullRequests} exact>
        <PullRequests />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODEWebhookNew} pageName={PAGE_NAME.CODEWebhookNew} exact>
        <WebhookNew />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODESearch} pageName={PAGE_NAME.CODERepository} exact>
        <Search />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODEWebhookDetails} pageName={PAGE_NAME.CODEWebhookDetails}>
        <WebhookDetails />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODEWebhooks} pageName={PAGE_NAME.CODEWebhooks} exact>
        <Webhooks />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODESettings} pageName={PAGE_NAME.CODESettings} exact>
        <Settings />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODERepositories} pageName={PAGE_NAME.CODERepositories} exact>
        <Repositories />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODECommits} pageName={PAGE_NAME.CODECommits}>
        <Commits />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODECommit} pageName={PAGE_NAME.CODECommits}>
        <Commit />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODEBranches} pageName={PAGE_NAME.CODEBranches} exact>
        <Branches />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODETags} pageName={PAGE_NAME.CODETags} exact>
        <Tags />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODEFileEdit} pageName={PAGE_NAME.CODEFileEdit}>
        <FileEdit />
      </RouteWithContext>

      <RouteWithContext path={routePaths.toCODERepository} pageName={PAGE_NAME.CODERepository}>
        <Repository />
      </RouteWithContext>
    </Switch>
  )
}

export const buildCODERoutePaths = (mode: NAV_MODE): Record<string, string | string[]> => {
  const repoPath = [
    codePathProps.accountId,
    codePathProps.orgIdentifier,
    codePathProps.projectIdentifier,
    codePathProps.repoName
  ].join('/')

  return {
    toCODERepositories: routes.toCODERepositories({
      space: [codePathProps.accountId, codePathProps.orgIdentifier, codePathProps.projectIdentifier].join('/'),
      mode
    }),
    toCODERepository: [
      routes.toCODERepository({
        repoPath,
        gitRef: codePathProps.gitRef,
        resourcePath: codePathProps.resourcePath,
        mode
      }),
      routes.toCODERepository({
        repoPath,
        gitRef: codePathProps.gitRef,
        mode
      }),
      routes.toCODERepository({ repoPath, mode })
    ],
    toCODEFileEdit: routes.toCODEFileEdit({
      repoPath,
      gitRef: codePathProps.gitRef,
      resourcePath: codePathProps.resourcePath,
      mode
    }),
    toCODECommits: routes.toCODECommits({
      repoPath,
      commitRef: codePathProps.commitRef,
      mode
    }),
    toCODECommit: routes.toCODECommit({
      repoPath,
      commitRef: codePathProps.commitRef,
      mode
    }),
    toCODEBranches: routes.toCODEBranches({ repoPath, mode }),
    toCODETags: routes.toCODETags({ repoPath, mode }),
    toCODECompare: routes.toCODECompare({
      repoPath,
      diffRefs: codePathProps.diffRefs,
      mode
    }),
    toCODEPullRequests: routes.toCODEPullRequests({ repoPath, mode }),
    toCODEPullRequest: [
      routes.toCODEPullRequest({
        repoPath,
        pullRequestId: codePathProps.pullRequestId,
        pullRequestSection: codePathProps.pullRequestSection,
        commitSHA: codePathProps.commitSHA,
        mode
      }),
      routes.toCODEPullRequest({
        repoPath,
        pullRequestId: codePathProps.pullRequestId,
        pullRequestSection: codePathProps.pullRequestSection,
        mode
      }),
      routes.toCODEPullRequest({
        repoPath,
        pullRequestId: codePathProps.pullRequestId,
        mode
      })
    ],
    toCODEWebhooks: routes.toCODEWebhooks({ repoPath, mode }),
    toCODEWebhookNew: routes.toCODEWebhookNew({ repoPath, mode }),
    toCODEWebhookDetails: routes.toCODEWebhookDetails({
      repoPath,
      webhookId: codePathProps.webhookId,
      mode
    }),
    toCODESearch: routes.toCODESearch({ repoPath, mode }),
    toCODESettings: [
      routes.toCODESettings({ repoPath, mode }),
      routes.toCODESettings({
        repoPath,
        settingSection: codePathProps.settingSection,
        settingSectionMode: codePathProps.settingSectionMode,
        ruleId: codePathProps.ruleId,
        mode
      }),
      routes.toCODESettings({
        repoPath,
        settingSection: codePathProps.settingSection,
        settingSectionMode: codePathProps.settingSectionMode,
        mode
      }),

      routes.toCODESettings({ repoPath, settingSection: codePathProps.settingSection, mode })
    ]
  }
}

const CODERedirect: React.FC = () => {
  const { scope, params } = useGetSelectedScope()
  const { selectedProject } = useAppStore()

  if (scope === Scope.PROJECT) {
    if (selectedProject?.modules?.includes(ModuleName.CODE)) {
      return (
        <Redirect
          to={routes.toCODERepositories({
            space: [params?.accountId, params?.orgIdentifier, params?.projectIdentifier].join('/')
          })}
        />
      )
    }
  }

  return <Redirect to={commonRoutes.toSettings({ orgIdentifier: params?.orgIdentifier, module: CODE })} />
}

const CODE = 'code'
