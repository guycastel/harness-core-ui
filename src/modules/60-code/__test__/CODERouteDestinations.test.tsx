/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */
import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { NAV_MODE } from '@modules/10-common/utils/routeUtils'
import mockImport from 'framework/utils/mockImport'
import type { StringsMap } from 'stringTypes'
import CODERouteDestinations, { buildCODERoutePaths } from '../CODERouteDestinations'

// eslint-disable-next-line jest-no-mock
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn()
  }),
  useRouteMatch: () => ({ params: { accountId: '1234', orgIdentifier: 'abc', projectIdentifier: 'xyz' } }),
  Switch: ({ children }: { children: JSX.Element }) => <>{children}</>,
  Redirect: () => <></>,
  useParams: () => ({ accountId: '1234' }),
  useLocation: () => ({ search: 'test' })
}))

jest.mock('@common/hooks', () => ({
  ...(jest.requireActual('@common/hooks') as any),
  useQueryParams: jest.fn().mockImplementation(() => ({ returnUrl: '/testing' }))
}))

jest.mock('@common/router/RouteWithContext/RouteWithContext', () => ({
  RouteWithContext: ({ children, path }: { children: JSX.Element; path: string }) => (
    <>
      {path}/{children}
    </>
  )
}))

jest.mock('framework/strings', () => ({
  useStrings: () => ({
    getString: (key: keyof StringsMap, _vars?: Record<string, any> | undefined) => key
  })
}))

jest.mock('@common/hooks/useTelemetry', () => ({
  useTelemetry: () => ({ identifyUser: jest.fn(), trackEvent: jest.fn() })
}))

jest.mock('@common/navigation/SideNavV2/SideNavV2.utils', () => ({
  useGetSelectedScope: () => ({
    scope: NAV_MODE.MODULE,
    params: { accountId: '1234', orgIdentifier: 'abc', projectIdentifier: 'xyz' }
  })
}))

jest.mock('framework/AppStore/AppStoreContext', () => ({
  useAppStore: () => ({ selectedProject: {} })
}))

jest.mock('../CodeApp', () => ({
  Branches: () => <div>code/Branches</div>,
  Commit: () => <div>code/Commit</div>,
  Commits: () => <div>code/Commits</div>,
  Compare: () => <div>code/Compare</div>,
  FileEdit: () => <div>code/FileEdit</div>,
  PullRequest: () => <div>code/PullRequest</div>,
  PullRequests: () => <div>code/PullRequests</div>,
  Repositories: () => <div>code/Repositories</div>,
  Repository: () => <div>code/Repository</div>,
  Search: () => <div>code/Search</div>,
  Settings: () => <div>code/Settings</div>,
  Tags: () => <div>code/Tags</div>,
  WebhookDetails: () => <div>code/WebhookDetails</div>,
  WebhookNew: () => <div>code/WebhookNew</div>,
  Webhooks: () => <div>code/Webhooks</div>
}))

mockImport('@common/hooks/useFeatureFlag', {
  useFeatureFlags: () => ({ CI_YAML_VERSIONING: true }),
  useFeatureFlag: () => false
})

describe('CODERouteDestinations Tests', () => {
  test('buildCODERoutePaths should render correct route paths for Module mode', async () => {
    const routePaths = buildCODERoutePaths(NAV_MODE.MODULE)

    expect(routePaths).toEqual({
      toCODEHome: '/account/:accountId/module/code/home',
      toCODEHomeTrial: '/account/:accountId/module/code/home/trial',
      toCODERepositories: '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos',
      toCODERepository: [
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/files/:gitRef*/~/:resourcePath*',
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/files/:gitRef*',
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName'
      ],
      toCODEFileEdit:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/edit/:gitRef*/~/:resourcePath*',
      toCODECommits:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/commits/:commitRef*',
      toCODECommit:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/commit/:commitRef*',
      toCODEBranches:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/branches',
      toCODETags:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/tags',
      toCODECompare:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/compare/:diffRefs*',
      toCODEPullRequests:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls',
      toCODEPullRequest: [
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/:pullRequestId/:pullRequestSection/:commitSHA',
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/:pullRequestId/:pullRequestSection',
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/:pullRequestId'
      ],
      toCODEWebhooks:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/webhooks',
      toCODEWebhookNew:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/webhooks/new',
      toCODEWebhookDetails:
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/webhooks/:webhookId',
      toCODESearch: [
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/search',
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/search'
      ],
      toCODESettings: [
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings',
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings/:settingSection/:ruleId/:settingSectionMode',
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings/:settingSection/:settingSectionMode',
        '/account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings/:settingSection'
      ]
    })
  })

  test('buildCODERoutePaths should render correct route paths for All mode', async () => {
    const routePaths = buildCODERoutePaths(NAV_MODE.ALL)

    expect(routePaths).toEqual({
      toCODEHome: '/account/:accountId/all/code/home',
      toCODEHomeTrial: '/account/:accountId/all/code/home/trial',
      toCODERepositories: '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos',
      toCODERepository: [
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/files/:gitRef*/~/:resourcePath*',
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/files/:gitRef*',
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName'
      ],
      toCODEFileEdit:
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/edit/:gitRef*/~/:resourcePath*',
      toCODECommits:
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/commits/:commitRef*',
      toCODECommit:
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/commit/:commitRef*',
      toCODEBranches:
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/branches',
      toCODETags: '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/tags',
      toCODECompare:
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/compare/:diffRefs*',
      toCODEPullRequests:
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls',
      toCODEPullRequest: [
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/:pullRequestId/:pullRequestSection/:commitSHA',
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/:pullRequestId/:pullRequestSection',
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/:pullRequestId'
      ],
      toCODEWebhooks:
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/webhooks',
      toCODEWebhookNew:
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/webhooks/new',
      toCODEWebhookDetails:
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/webhooks/:webhookId',
      toCODESearch: [
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/search',
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/search'
      ],
      toCODESettings: [
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings',
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings/:settingSection/:ruleId/:settingSectionMode',
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings/:settingSection/:settingSectionMode',
        '/account/:accountId/all/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings/:settingSection'
      ]
    })
  })

  test('CODERouteDestinations should render correct routes', async () => {
    const { container } = render(<>{CODERouteDestinations()}</>)
    await waitFor(() => {
      expect(container).toMatchSnapshot()
    })
  })
})
