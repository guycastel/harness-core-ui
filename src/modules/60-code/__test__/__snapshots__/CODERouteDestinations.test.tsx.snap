// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CODERouteDestinations Tests CODERouteDestinations should render correct routes 1`] = `
<div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier
  /account/:accountId/module/code/orgs/:orgIdentifier
  /account/:accountId/module/code
  /
  /account/:accountId/module/code/home/trial
  /
  <div
    class="StyledProps--font StyledProps--main body"
    style="background: transparent url(test-file-stub) no-repeat;"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
    >
      <h1
        class="StyledProps--font StyledProps--font-h1 StyledProps--main StyledProps--font-weight-bold StyledProps--font-size-large StyledProps--color StyledProps--color-black100"
      >
        common.purpose.code.title
      </h1>
      <div
        class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--padding-bottom-xxlarge"
          style="width: 500px;"
        >
          code.homepageHeading
        </p>
        <a
          class="learnMore"
          href="https://developer.harness.io/docs/code-repository"
          rel="noreferrer"
          target="_blank"
        >
          code.learnMore
        </a>
        <button
          aria-label="common.startFreePlan"
          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
          style="width: 300px; height: 45px;"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            common.startFreePlan
          </span>
        </button>
      </div>
    </div>
  </div>
  /account/:accountId/module/code/home
  /
  <div
    class="StyledProps--font StyledProps--main container"
    style="background: transparent url(test-file-stub) no-repeat;"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
    >
      <h1
        class="StyledProps--font StyledProps--font-h1 StyledProps--main StyledProps--font-weight-bold StyledProps--font-size-large StyledProps--color StyledProps--color-black100"
      >
        common.purpose.code.title
      </h1>
      <p
        class="StyledProps--font StyledProps--main subTitle StyledProps--color"
      >
        code.homepageHeading
      </p>
      <a
        class="bp3-button Button--button StyledProps--font StyledProps--main link StyledProps--color StyledProps--font StyledProps--font-size-normal Button--with-current-color Button--link"
        href="https://developer.harness.io/docs/category/get-started-with-code"
        rel="noreferrer noopener"
        role="button"
        tabindex="0"
        target="_blank"
      >
        <span
          class="bp3-button-text"
        >
          code.learnMore
        </span>
      </a>
      <div
        class="Layout--horizontal Layout--layout-spacing-large StyledProps--main StyledProps--flex"
      >
        <button
          class="bp3-button bp3-large Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-primary"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            createProject
          </span>
        </button>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--color StyledProps--color-black"
        >
          orSelectExisting
        </p>
        <span
          style="flex-grow: 1;"
        />
      </div>
    </div>
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/compare/:diffRefs*
  /
  <div>
    code/Compare
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/:pullRequestId/:pullRequestSection/:commitSHA
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/:pullRequestId/:pullRequestSection
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls/:pullRequestId
  /
  <div>
    code/PullRequest
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/pulls
  /
  <div>
    code/PullRequests
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/webhooks/new
  /
  <div>
    code/WebhookNew
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/search
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/search
  /
  <div>
    code/Search
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/webhooks/:webhookId
  /
  <div>
    code/WebhookDetails
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/webhooks
  /
  <div>
    code/Webhooks
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings/:settingSection/:ruleId/:settingSectionMode
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings/:settingSection/:settingSectionMode
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/settings/:settingSection
  /
  <div>
    code/Settings
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos
  /
  <div>
    code/Repositories
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/commits/:commitRef*
  /
  <div>
    code/Commits
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/commit/:commitRef*
  /
  <div>
    code/Commit
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/branches
  /
  <div>
    code/Branches
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/tags
  /
  <div>
    code/Tags
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/edit/:gitRef*/~/:resourcePath*
  /
  <div>
    code/FileEdit
  </div>
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/files/:gitRef*/~/:resourcePath*
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName/files/:gitRef*
  /account/:accountId/module/code/orgs/:orgIdentifier/projects/:projectIdentifier/repos/:repoName
  /
  <div>
    code/Repository
  </div>
</div>
`;
